codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -161,29 +161,17 @@ public B gracefulShutdownTimeoutMillis(long gracefulShutdownTimeoutMillis) {
         }
 
         /**
-         * Determine if the encoder should queue frames to honor the value set by
-         * {@link #encoderMaxConcurrentStreams(int)}.
-         */
-        public B encoderEnforceMaxConcurrentStreams(boolean encoderEnforceMaxConcurrentStreams) {
-            encoderMaxConcurrentStreams = -1;
-            return thisB();
-        }
-
-        private boolean encoderEnforceMaxConcurrentStreams() {
-            return encoderMaxConcurrentStreams >= 0;
-        }
-
-        /**
          * How many initial streams are allowed to exists concurrently. Frames will be queued if they would result in
          * creating a stream which would cause the number of existing streams to exceed this number.
-         * @see #encoderEnforceMaxConcurrentStreams(boolean)
+         *
+         * Use {@code -1} if you not want to enforce max concurrent streams.
          */
         public B encoderMaxConcurrentStreams(int encoderMaxConcurrentStreams) {
             // This bounds are enforced here because the builder makes assumptions about its valid range to determine
             // if it should be used.
-            if (encoderMaxConcurrentStreams < 0) {
+            if (encoderMaxConcurrentStreams < -1) {
```

Comment:
@normanmaurer - Actually I think we should step back a second. Now that we have all the context in 1 the builder (settings, and creation of the encoder) I wonder if we should use the following approach:

- remove `private int encoderMaxConcurrentStreams`.
- Add a member and method:
```java
private boolean encoderBuffer = true;
public B encoderBufferMaxConcurrentStreams(boolean buffer) {
  encoderBuffer = buffer;
  return thisB();
}
```
- In the `build(Http2Connection)` method have the following:
```java
if (encoderBuffer) {
Long maxConcurrentStreams = initialSettings.maxConcurrentStreams();
encoder = maxConcurrentStreams == null ? new StreamBufferingEncoder(encoder) : new StreamBufferingEncoder(encoder, /*do appropriate type cast / bounds check */ maxConcurrentStreams);
}
```

@nmittler - WDYT?